### DISCUSSION TITLE
run-a-command-against-a-group-of-devices is running, but not getting result

### DISCUSSION THREAD

**Discussion by ninja5244** [CrowdStrike/psfalcon#414] [Q&A]
I'm trying to run-a-command-against-a-group-of-devices using below script, and it is running and giving me the queue status. but not getting the result. For example if i'm trying to create a folder using PowerShell against devices in a group, the script is running but the folder is not creating. Below is my script, if anyone can find the problem in, kindly let me know. thanks in advance
```
using module @{ModuleName='PSFalcon';ModuleVersion ='2.2'}
param(
  [Parameter(Mandatory,Position=1)]
  [string]$GroupName,
  [Parameter(Mandatory,Position=2)]
  [string]$Command,
  [Parameter(Position=3)]
  [string]$Argument,
  [boolean]$QueueOffline
)

$GroupId = Get-FalconHostGroup -Filter "name:'$($GroupName.ToLower())'"
if (!$GroupId) { throw "No host group found matching '$($GroupName.ToLower())'." }
$Param = @{
  GroupId = $GroupId 
  Command = $Command
}
$HostIds = Get-FalconHostGroupMember -Id $GroupId -All
Invoke-FalconRTR -Command runscript  -HostIds $HostIds -QueueOffline $true 
```

**Comment by bk-cs** [comment]
> I'm trying to run-a-command-against-a-group-of-devices using below script, and it is running and giving me the queue status. but not getting the result. For example if i'm trying to create a folder using PowerShell against devices in a group, the script is running but the folder is not creating. Below is my script, if anyone can find the problem in, kindly let me know. thanks in advance

The script uses `-QueueOffline $true`. If all the devices in your group are currently offline, the command will be queued and it will attempt to run on the devices once they re-connect to the Falcon cloud, as long as that's within 7 days since the command was added to the queue.

You can also use `Invoke-FalconRtr` by itself and target a host group by it's id instead of using this script. 

What does the output look like when you run the script? Based on your description, it sounds like it's running as expected and you're not getting results because none of your target hosts are currently online.
